File: sources/LiquidSense Release/me/aquavit/liquidsense/module/modules/movement/SafeWalk.java
package me.aquavit.liquidsense.module.modules.movement;

import me.aquavit.liquidsense.event.EventTarget;
import me.aquavit.liquidsense.event.events.MoveEvent;
import me.aquavit.liquidsense.module.Module;
import me.aquavit.liquidsense.module.ModuleCategory;
import me.aquavit.liquidsense.module.ModuleInfo;
import me.aquavit.liquidsense.value.BoolValue;

@ModuleInfo(name = "SafeWalk", description = "Prevents you from falling down as if you were sneaking.", category = ModuleCategory.MOVEMENT)
public class SafeWalk extends Module {
    private BoolValue airSafeValue = new BoolValue("AirSafe", false);

    @EventTarget
    public void onMove(MoveEvent event){
        if (airSafeValue.get() || mc.thePlayer.onGround)
            event.setSafeWalk(true);
    }
}

File: sources/Kevin Reborn/kevin/module/modules/movement/SafeWalk.kt
/*
 * This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package kevin.module.modules.movement

import kevin.event.EventTarget
import kevin.event.MoveEvent
import kevin.module.BooleanValue
import kevin.module.Module
import kevin.module.ModuleCategory
import net.minecraft.item.ItemBlock

class SafeWalk : Module("SafeWalk", "Prevents you from falling down as if you were sneaking.", category = ModuleCategory.MOVEMENT) {
    private val airSafeValue = BooleanValue("AirSafe", false)
    private val onlyBlock by BooleanValue("OnlyBlockInHand", false)

    @EventTarget
    fun onMove(event: MoveEvent) {
        if (onlyBlock) {
            val heldItem = mc.thePlayer?.heldItem ?: return
            if (heldItem.item !is ItemBlock) return
        }
        if (airSafeValue.get() || mc.thePlayer!!.onGround)
            event.isSafeWalk = true
    }
}